PlayerSkeleton.java

```diff
@@ -40,10 +40,89 @@ private static double evaluate(SerializedState s, double[] weights) {
 		valuation += s.cleared * weights[0];
 		valuation += getHighestCol(s.field) * weights[1];
 		valuation += getLowestCol(s.field) * weights[2];
+		valuation +=  getHolesCount(s.field, s.tops) * weights[3];
+		valuation += getBlockadeCount(s.field, s.tops) * weights[4];
+		valuation += getParityCount(s.field) * weights[5] ;
+		valuation += getTotalHeightDiff(s.tops) * weights[6];
 
 		return valuation;
 	}
 
+	private static int getHolesCount(int[][] field, int[] tops) {

```

Comment:
Does this count an empty column of 10 covered by 1 block is as 1 hole? I think we should treat it as 10 instead. The simple way is to just count the number of 0s from `top` to the floor.